anaDACScan.py

```diff
@@ -0,0 +1,236 @@
+    r.TH1.SetDefaultSumw2(False)
+    r.gROOT.SetBatch(True)
+    r.gStyle.SetOptStat(1111111)
+
+    dacScanFile = r.TFile(args.infilename)
+
+    import numpy as np
+    import root_numpy as rp
+    
+    list_bNames = ['link']
+    ohArray = rp.tree2array(tree=dacScanFile.dacScanTree, branches=list_bNames)
+    ohArray = np.unique(ohArray['link'])
+    
+    dataPath = os.getenv("DATA_PATH") 
+    
+    if len(args.infilename.split('/')) > 1 and len(args.infilename.split('/')[len(args.infilename.split('/')) - 2].split('.')) == 5:
+        scandate = args.infilename.split('/')[len(args.infilename.split('/')) - 2]
+    else:    
+        scandate = 'noscandate'
+
+    for oh in ohArray:
+        os.system("mkdir -p "+ dataPath+"/"+chamber_config[oh]+"/dacScans/"+scandate)
+            
+    # Determine which DAC was scanned and against which ADC
+    nameX = ""
+    nameY = ""
+    for event in dacScanFile.dacScanTree:
+        nameX = str(event.nameX.data())
+        nameY = str(event.nameY.data())
+        break # all entries will be the same
+
+    if nameY not in ['ADC0', 'ADC1']:
+        print("Error: unexpected value of nameY: '%s'"%nameY)
+        exit(1)
+
+    calInfo = {}
+
+    if args.calFileList != None:
+        for line in open(args.calFileList):
+            line = line.strip(' ').strip('\n')
+            first = line.split(' ')[0].strip(' ')
+            second = line.split(' ')[1].strip(' ')
+            tuple_calInfo = parseCalFile(second)
+            calInfo[int(first)] = {'slope' : tuple_calInfo[0], 'intercept' : tuple_calInfo[1]}
+
+    #for each OH, check if calibration files were provided, if not search for the calFile in the $DATA_PATH, if it is not there, then skip that OH for the rest of the script
+    for oh in ohArray:
+        if oh not in calInfo.keys():
+            calDacCalFile = "{0}/{1}/calFile_calDac_{1}.txt".format(dataPath,chamber_config[oh])
+            calDacCalFileExists = os.path.isfile(calDacCalFile)
+            if not calDacCalFileExists:
+                print("Skipping OH{0}, detector {1}, missing CFG_CAL_DAC Calibration file:\n\t{2}".format(
+                    oh,
+                    chamber_config[oh],
+                    calDacCalFile))
+                ohArray = np.delete(ohArray,(oh))
+
+
+    if len(ohArray) == 0:
+        print('No OHs with a calFile, exiting.')
+        exit(1)
+           
+    # Initialize nested dictionaries
+    dict_dacScanResults = nesteddict()
+    dict_dacScanFuncs = nesteddict()
+
+    # Initialize a TGraphErrors for each vfat
+    for oh in ohArray:
+        for vfat in range(0,24):
+             dict_dacScanResults[oh][vfat] = r.TGraphErrors()
+             #the reversal of x and y is intended - we want to plot the nameX variable on the y-axis and the nameY variable on the x-axis
+             dict_dacScanResults[oh][vfat].GetXaxis().SetTitle(nameY)
+             dict_dacScanResults[oh][vfat].GetYaxis().SetTitle(nameX)
```

Comment:
It would also be good to have the actual DAC scan plotted e.g. `ADCX vs <reg>`.  Looking at the raw data helps to understand if the further results should be considered valid.

Reply:
I would also clarify the comment in line 158; e.g. why is this reversal being done?  From here it's not illustrated tha the goal is to obtain nominal value of register `nameX`.